x-pack/plugins/fleet/public/components/agent_enrollment_flyout/index.tsx

```diff
@@ -60,6 +64,24 @@ export const AgentEnrollmentFlyout: React.FunctionComponent<Props> = ({
   const [policyId, setSelectedPolicyId] = useState(agentPolicy?.id);
   const [isFleetServerPolicySelected, setIsFleetServerPolicySelected] = useState<boolean>(false);
 
+  // loading the latest agentPolicies for add agent flyout
+  const {
+    data: agentPoliciesData,
+    isLoading: isLoadingAgentPolicies,
+    resendRequest: refreshAgentPolicies,
+  } = useGetAgentPolicies({
```

Comment:
I decided to load/refresh agent policies here in AgentEnrollmentFlyout.
The reason is that AgentEnrollmentFlyout is being called from 7 places, from which 4 are used without agentPolicies, 3 with agentPolicies. It would be too much duplication to reload agentPolicies from all those caller sites.
AgentEnrollmentFlyout Props should be cleaned up so that agentPolicies is not passed as a prop.

Reply:
What do you think of creating a hook like `useAgentEnrollmentFlyoutData` that contains this and the `useMemo` below, to start separating more business and presentation in this component?